test/test_ltp.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import unittest
+from mock import PropertyMock, Mock
+from ltp import LtpLogs
+
+
+TEST_LOG = """
+test_1: some log 1
+test_1: some log 2
+test_1: some log 3
+test_3: some log 1
+test_3: some log 2
+test_3: some log 3
+"""
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+
+class LtpLogsPluginTest(unittest.TestCase):
+    def setUp(self):
+        self.plugin = LtpLogs()
+
+    def test_postprocess_testrun(self):
+        test_mock_1 = Mock()
+        test_mock_1_name = PropertyMock(return_value="test_1")
+        type(test_mock_1).name = test_mock_1_name
+        test_mock_1_log = PropertyMock()
+        type(test_mock_1).log = test_mock_1_log
+        test_mock_2 = Mock()
+        test_mock_2_name = PropertyMock(return_value="test_2")
+        type(test_mock_2).name = test_mock_2_name
+        test_mock_2_log = PropertyMock()
+        type(test_mock_2).log = test_mock_2_log
+        test_mock_3 = Mock()
+        test_mock_3_name = PropertyMock(return_value="test_3")
+        type(test_mock_3).name = test_mock_3_name
+        test_mock_3_log = PropertyMock()
+        type(test_mock_3).log = test_mock_3_log
+        testrun_mock = Mock()
+        testrun_log_mock = PropertyMock(return_value=TEST_LOG)
+        type(testrun_mock).log_file = testrun_log_mock
+        testrun_mock.tests = Mock()
+        testrun_mock.tests.filter.return_value = [test_mock_1, test_mock_2, test_mock_3]
+        self.plugin.postprocess_testrun(testrun_mock)
+        testrun_mock.tests.filter.assert_called()
+        testrun_log_mock.assert_called()
+        test_mock_1.save.assert_called()
+        test_mock_1_name.assert_called()
+        test_mock_1_log.assert_called()
+        test_mock_2.save.assert_not_called()
+        test_mock_2_name.assert_called()
+        test_mock_2_log.assert_not_called()
+        test_mock_3.save.assert_called()
+        test_mock_3_name.assert_called()
+        test_mock_3_log.assert_called()
```

Comment:
suggestion to make this more readable:

* drop "mock" from variable names
* separate the setup of different objects with empty lines
* separate assertions against different objects with empty lines

this would look like this:

```
        test1 = Mock()
        test1_name = PropertyMock(return_value='test_1')
        type(test1).name = test1_name
        test1_log = PropertyMock()
        type(test1).log = test1_log

        test2 = Mock()
        test2_name = PropertyMock(return_value='test_2')
        type(test2).name = test2_name
        test2_log = PropertyMock()
        type(test2).log = test2_log

        test3 = Mock()
        test3_name = PropertyMock(return_value='test_3')
        type(test3).name = test3_name
        test3_log = PropertyMock()
        type(test3).log = test3_log

        testrun = Mock()
        log_file = PropertyMock(return_value=TEST_LOG)
        type(testrun).log_file = log_file
        testrun.tests.filter.return_value = [test1, test2, test3]
        self.plugin.postprocess_testrun(testrun)

        self.plugin.postprocess_testrun(testrun)
        testrun.tests.filter.assert_called()
        log_file.assert_called()

        test1_name.assert_called()
        test1_log.assert_called()
        test1.save.assert_called()

        # test2 not present in the test run log
        test2_name.assert_called()
        test2_log.assert_not_called()
        test2.save.assert_not_called()

        test3_name.assert_called()
        test3_log.assert_called()
        test3.save.assert_called()
```